src/opentelemetry/_opamp/messages.py

```diff
@@ -0,0 +1,113 @@
+        return PB2AnyValue(int_value=value)
+    if isinstance(value, float):
+        return PB2AnyValue(double_value=value)
+    if isinstance(value, str):
+        return PB2AnyValue(string_value=value)
+    if isinstance(value, bytes):
+        return PB2AnyValue(bytes_value=value)
+    # TODO: handle sequence and mapping?
+    raise ValueError(f"Invalid type {type(value)} of value {value}")
+
+
+def _encode_attributes(attributes: dict[str, AnyValue]):
+    return [PB2KeyValue(key=key, value=_encode_value(value)) for key, value in attributes.items()]
+
+
+def _build_agent_description(
+    identifying_attributes: dict[str, AnyValue],
+    non_identifying_attributes: dict[str, AnyValue] | None = None,
+) -> opamp_pb2.AgentDescription:
+    identifying_attrs = _encode_attributes(identifying_attributes)
+    non_identifying_attrs = _encode_attributes(non_identifying_attributes) if non_identifying_attributes else None
+    return opamp_pb2.AgentDescription(
+        identifying_attributes=identifying_attrs, non_identifying_attributes=non_identifying_attrs
+    )
+
+
+def _build_presentation_message(
+    instance_uid: bytes, sequence_num: int, agent_description: opamp_pb2.AgentDescription, capabilities: int
+) -> opamp_pb2.AgentToServer:
+    command = opamp_pb2.AgentToServer(
+        instance_uid=instance_uid,
+        sequence_num=sequence_num,
+        agent_description=agent_description,
+        capabilities=capabilities,
+    )
+    return command
+
+
+def _build_heartbeat_message(instance_uid: bytes, sequence_num: int) -> opamp_pb2.AgentToServer:
+    command = opamp_pb2.AgentToServer(instance_uid=instance_uid, sequence_num=sequence_num)
+    return command
+
+
+def _build_agent_disconnect_message(instance_uid: bytes, sequence_num: int) -> opamp_pb2.AgentToServer:
+    command = opamp_pb2.AgentToServer(
+        instance_uid=instance_uid,
+        sequence_num=sequence_num,
+        agent_disconnect=opamp_pb2.AgentDisconnect(),
+    )
+    return command
+
+
+def _encode_message(data: opamp_pb2.AgentToServer) -> bytes:
+    return data.SerializeToString()
+
+
+def _decode_remote_config(remote_config: opamp_pb2.AgentRemoteConfig) -> Generator[tuple[str, dict[str, AnyValue]]]:
+    for config_file_name, config_file in remote_config.config.config_map.items():
+        if config_file.content_type in ("application/json", "text/json"):
+            try:
+                body = config_file.body.decode()
+                config_data = json.loads(body)
+            except (UnicodeDecodeError, json.JSONDecodeError) as exc:
+                raise OpAMPRemoteConfigDecodeException(
+                    f"Failed to decode {config_file} with content type {config_file.content_type}: {exc}"
+                )
+                continue
+
+            yield config_file_name, config_data
+        else:
+            raise OpAMPRemoteConfigParseException(
+                f"Cannot parse {config_file_name} with content type {config_file.content_type}"
```

Comment:
nit: Not sure if I read this correctly, but if the OpAMP server returns *any* config file with a content-type other than one of the two JSON ones, we'll throw here, and eventually get:

```python
logger.warning("Job %r handler failed with: %s", job.payload, exc)
```

Reply:
Yes, do you suggest to ignore errors on not elastic filenames? I don't expect any though

Reply:
> I don't expect any though

I called this a "nit" because I'm hunting for edge cases that are unlikely.


> do you suggest to ignore errors on not _elastic_ filenames?

That's what my implementation is doing:
https://github.com/elastic/elastic-otel-node/blob/682bdc2444e9da6a25adb2ea2d03fd11cfe89783/packages/opentelemetry-node/lib/central-config.js#L89-L102

It only ever looks at the `elastic` entry in the configMap.

Reply:
I'll try to upstream the client so I cannot add elastic specific cases when decoding messages, if that would become a problem in practice I can revise it later I guess.